classes/local/files/options_file_service.php

```diff
@@ -0,0 +1,246 @@
+<?php
+// This file is part of the QuestionPy Moodle plugin - https://questionpy.org
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace qtype_questionpy\local\files;
+
+use coding_exception;
+use context_user;
+use DateTimeImmutable;
+use file_exception;
+use moodle_exception;
+use moodle_url;
+use qtype_questionpy_question;
+use stored_file;
+use stored_file_creation_exception;
+
+/**
+ * Handles files uploaded by trainers in question options.
+ *
+ * @package    qtype_questionpy
+ * @author     Maximilian Haye
+ * @copyright  2025 TU Berlin, innoCampus {@link https://www.questionpy.org}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class options_file_service implements handles_qpy_url_type {
+    // TODO: Support subdirectories.
+
+    // TODO: When a file is deleted while editing a question, we do not currently delete it from the permanent file area (it just
+    // gets hidden because it's removed from the metadata). We probably should do so in the future, but we must make sure that
+    // previous question versions remain functional _if_ they still exist.
+
+    /** @var string */
+    public const FILEAREA_UPLOADS = 'options';
+
+    /**
+     * Saves all the files in the given draft item to the permanent file area for the given question.
+     *
+     * @param int $contextid Context id of the question (NOT the draft area).
+     * @param int $questionid
+     * @param int $userid User whose draft area should be used, which is most likely the current user.
+     * @param int $draftitemid
+     * @throws file_exception
+     * @throws stored_file_creation_exception
+     * @throws coding_exception
+     */
+    public function save_draft_area_files(int $contextid, int $questionid, int $userid, int $draftitemid): void {
```

Comment:
Ich bin mir nicht sicher, ob ich noch irgendwas in Moodle übersehen habe, aber ich glaube, dass wir die Dateien noch selber validieren müssen, weil `file_save_draft_area_files` nicht genutzt wird.

Reply:
Passiert jetzt hier: 18bdaef392d593dcfea9c0d2464d24cf90bd6b52. Wird in den `on_export` aufgerufen, also nicht nur beim speichern, auch, bevor es an den QPy-Server geschickt wird.

In meinen Tests konnte ich da als freundlicher Nutzer aber auch gar nicht hinkommen, Moodle blockt das vorher mit einer schönen Fehlermeldung. Daher nur `coding_exceptions`.